wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -23,129 +23,101 @@ import WireDataModel
-                    }
-                ).first
-        else {
-            return isE2EIdentityEnabled ? .notActivated : .none
-        }
-        return status
+    var isCopyEnabled: Bool {
+        return Settings.isClipboardEnabled
     }
 
-    var isCertificateExpiringSoon: Bool {
+    var isCertificateExpiringSoon: Bool? {
         guard let certificate = e2eIdentityCertificate else {
-            return false
+            return nil
         }
-        return certificate.expiryDate < Date.now + .oneDay + .oneDay
+        return certificate.shouldUpdate(with: gracePeriod)
     }
 
     @Published
     var e2eIdentityCertificate: E2eIdentityCertificate?
-    @Published
-    var mlsThumbprint: String?
-    private var actionsHandler: any DeviceDetailsViewActions
-    var isCopyEnabled: Bool {
-        return Settings.isClipboardEnabled
-    }
     @Published var isRemoved: Bool = false
     @Published var isProteusVerificationEnabled: Bool = false
     @Published var isActionInProgress: Bool = false
     @Published var proteusKeyFingerprint: String = ""
+    @Published var isE2eIdentityEnabled = false
+
+    private var actionsHandler: any DeviceDetailsViewActions
 
     init(
         title: String,
         addedDate: String,
         proteusID: String,
+        mlsThumbprint: String?,
         isProteusVerificationEnabled: Bool,
         actionsHandler: any DeviceDetailsViewActions,
-        userSession: UserSession,
-        getUserClientFingerprint: GetUserClientFingerprintUseCaseProtocol,
         userClient: UserClient,
-        isSelfClient: Bool
+        isSelfClient: Bool,
+        gracePeriod: TimeInterval
     ) {
         self.title = title
         self.addedDate = addedDate
         self.proteusID = proteusID
+        self.mlsThumbprint = mlsThumbprint
         self.isProteusVerificationEnabled = isProteusVerificationEnabled
         self.actionsHandler = actionsHandler
-        self.userSession = userSession
-        self.getUserClientFingerprint = getUserClientFingerprint
         self.userClient = userClient
         self.isSelfClient = isSelfClient
+        self.gracePeriod = gracePeriod
         self.actionsHandler.isProcessing = {[weak self] isProcessing in
             DispatchQueue.main.async {
                 self?.isActionInProgress = isProcessing
             }
         }
     }
 
-    func fetchFingerPrintForProteus() async {
-        DispatchQueue.main.async {
-            self.isActionInProgress = true
-        }
-        guard let data = await getUserClientFingerprint.invoke(userClient: userClient),
-                let fingerPrint = String(data: data, encoding: .utf8) else {
-            return
-        }
-        DispatchQueue.main.async {
-            self.proteusKeyFingerprint = fingerPrint.splitStringIntoLines(charactersPerLine: 16).uppercased()
-            self.isActionInProgress = false
-        }
+    @MainActor
+    func getE2eCertificate() async {
+        self.isActionInProgress = true
```

Comment:
Just a note:
The method can be suspended.
If I am not mistaken, in the meantime another method could run from start to end, which would result in `isActionInProgress` being wrong.
Depending on what the purpose of the property is, this should be considered. For showing a progress indicator it might not matter.

Reply:
This is just used for progress